## 데이터 불러오기

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
df_info = pd.read_csv('/content/drive/MyDrive/Streamlit 해커톤/data/info.csv')
df_info.head(3)

index                                               link  \
0      0  https://ottogi.co.kr/product/product_view.asp?...   
1      1  https://ottogi.co.kr/product/product_view.asp?...   
2      2  https://ottogi.co.kr/product/product_view.asp?...   

                                                body  
0  * 짜슐랭 3스타가이드하나. 고온 로스팅으로 맛을 살린 진한 짜장둘. 파와 양파 기...  
1  1. 바삭하고 고소한 면발에 바베큐맛 스프를 골고루 뿌려서 먹는 부숴 먹는 라면스낵...  
2  1. 120kcal 라는 낮은 칼로리의 제품입니다.2. 담백하고 얼큰한 맛을 즐길 ...

In [ ]:
df_name = pd.read_csv('/content/drive/MyDrive/Streamlit 해커톤/data/crawling_link.csv')
df_name.head(3)

index      title                                               link  \
0      0        짜슐랭  https://ottogi.co.kr/product/product_view.asp?...   
1      1  뿌셔뿌셔 바베큐맛  https://ottogi.co.kr/product/product_view.asp?...   
2      2    컵누들 우동맛  https://ottogi.co.kr/product/product_view.asp?...   

                                               image       category  
0  https://ottogi.co.kr/pds/product/2022-05-16_10...      라면류 > 짜슐랭  
1  https://ottogi.co.kr/pds/product/2023-05-30_64...  라면스낵류 > 라면스낵류  
2  https://ottogi.co.kr/pds/product/2023-05-10_22...      라면류 > 컵누들

In [ ]:
# 데이터를 합치기 위해 중복 열 제거
df_name = df_name.drop(['index', 'link'], axis=1)

In [ ]:
# 두 데이터 합치기
df = pd.concat([df_name, df_info], axis=1)
df

title                                              image  \
0            짜슐랭  https://ottogi.co.kr/pds/product/2022-05-16_10...   
1      뿌셔뿌셔 바베큐맛  https://ottogi.co.kr/pds/product/2023-05-30_64...   
2        컵누들 우동맛  https://ottogi.co.kr/pds/product/2023-05-10_22...   
3        진라면 매운맛  https://ottogi.co.kr/pds/product/2022-05-16_10...   
4      쇠고기미역국 라면  https://ottogi.co.kr/pds/product/2020-08-26_55...   
..           ...                                                ...   
995     한입쏙쏙 미트볼  https://ottogi.co.kr/pds/product/2022-03-16_15...   
996  달콤한 데리야끼 치킨  https://ottogi.co.kr/pds/product/2020-03-26_50...   
997      화끈한 닭강정  https://ottogi.co.kr/pds/product/2020-03-26_50...   
998  떠먹는 컵피자 불고기  https://ottogi.co.kr/pds/product/2018-06-12_95...   
999  헬로베지_채소가득카레  https://ottogi.co.kr/pds/product/2022-07-06_51...   

           category  index                                               link  \
0         라면류 > 짜슐랭      0  https://ottogi.co.kr/product/product_view.asp?...   
1     라면스낵류 > 라면스낵류      1  https://ottogi.co.kr/product/product_view.asp?...   
2         라면류 > 컵누들      2  https://ottogi.co.kr/product/product_view.asp?...   
3     라면류 > 진라면 매운맛      3  https://ottogi.co.kr/product/product_view.asp?...   
4    라면류 > 쇠고기미역국라면      4  https://ottogi.co.kr/product/product_view.asp?...   
..              ...    ...                                                ...   
995   레토르트/간편식 > 렌지    995  https://ottogi.co.kr/product/product_view.asp?...   
996   레토르트/간편식 > 렌지    996  https://ottogi.co.kr/product/product_view.asp?...   
997   레토르트/간편식 > 렌지    997  https://ottogi.co.kr/product/product_view.asp?...   
998  냉장/냉동식품 > 냉동피자    998  https://ottogi.co.kr/product/product_view.asp?...   
999   레토르트/간편식 > 3분    999  https://ottogi.co.kr/product/product_view.asp?...   

                                                  body  
0    * 짜슐랭 3스타가이드하나. 고온 로스팅으로 맛을 살린 진한 짜장둘. 파와 양파 기...  
1    1. 바삭하고 고소한 면발에 바베큐맛 스프를 골고루 뿌려서 먹는 부숴 먹는 라면스낵...  
2    1. 120kcal 라는 낮은 칼로리의 제품입니다.2. 담백하고 얼큰한 맛을 즐길 ...  
3              쫄깃하고 부드러운 면발에 진한국물과 얼큰한 양념이 잘 조화된 맛입니다.  
4    쇠고기양지와 사골을 우려낸 진한 육수에 쫄깃하고 부드러운 면발과남해안산 청정미역을 ...  
..                                                 ...  
995  1. 새콤 달콤 맛있는 토마토 소스로 완성한 미트볼, 식이섬유가 풍부하고 두뇌건강에...  
996  1. 달콤하고 짭조름한 데리야끼 소스로 맛있게 조리한 데리야끼 치킨!  2. 달콤짭...  
997  1. 이제는 매운맛이 대세! 매운 고추와 함께 사천요리에 사용되는 두반장과 고추기름...  
998             자연치즈 듬뿍간편하게 떠먹는 컵피자 !누구나 좋아하는 불고기 가득 !  
999  1.비건 브랜드인 HELLO VEGGIE 적용-영국 비건 소사이어티 인증2.8가지 ...  

[1000 rows x 6 columns]

In [ ]:
# 열 순서 변경
df = df[['index', 'title', 'category', 'body', 'image', 'link']]
df

index        title        category  \
0        0          짜슐랭       라면류 > 짜슐랭   
1        1    뿌셔뿌셔 바베큐맛   라면스낵류 > 라면스낵류   
2        2      컵누들 우동맛       라면류 > 컵누들   
3        3      진라면 매운맛   라면류 > 진라면 매운맛   
4        4    쇠고기미역국 라면  라면류 > 쇠고기미역국라면   
..     ...          ...             ...   
995    995     한입쏙쏙 미트볼   레토르트/간편식 > 렌지   
996    996  달콤한 데리야끼 치킨   레토르트/간편식 > 렌지   
997    997      화끈한 닭강정   레토르트/간편식 > 렌지   
998    998  떠먹는 컵피자 불고기  냉장/냉동식품 > 냉동피자   
999    999  헬로베지_채소가득카레   레토르트/간편식 > 3분   

                                                  body  \
0    * 짜슐랭 3스타가이드하나. 고온 로스팅으로 맛을 살린 진한 짜장둘. 파와 양파 기...   
1    1. 바삭하고 고소한 면발에 바베큐맛 스프를 골고루 뿌려서 먹는 부숴 먹는 라면스낵...   
2    1. 120kcal 라는 낮은 칼로리의 제품입니다.2. 담백하고 얼큰한 맛을 즐길 ...   
3              쫄깃하고 부드러운 면발에 진한국물과 얼큰한 양념이 잘 조화된 맛입니다.   
4    쇠고기양지와 사골을 우려낸 진한 육수에 쫄깃하고 부드러운 면발과남해안산 청정미역을 ...   
..                                                 ...   
995  1. 새콤 달콤 맛있는 토마토 소스로 완성한 미트볼, 식이섬유가 풍부하고 두뇌건강에...   
996  1. 달콤하고 짭조름한 데리야끼 소스로 맛있게 조리한 데리야끼 치킨!  2. 달콤짭...   
997  1. 이제는 매운맛이 대세! 매운 고추와 함께 사천요리에 사용되는 두반장과 고추기름...   
998             자연치즈 듬뿍간편하게 떠먹는 컵피자 !누구나 좋아하는 불고기 가득 !   
999  1.비건 브랜드인 HELLO VEGGIE 적용-영국 비건 소사이어티 인증2.8가지 ...   

                                                 image  \
0    https://ottogi.co.kr/pds/product/2022-05-16_10...   
1    https://ottogi.co.kr/pds/product/2023-05-30_64...   
2    https://ottogi.co.kr/pds/product/2023-05-10_22...   
3    https://ottogi.co.kr/pds/product/2022-05-16_10...   
4    https://ottogi.co.kr/pds/product/2020-08-26_55...   
..                                                 ...   
995  https://ottogi.co.kr/pds/product/2022-03-16_15...   
996  https://ottogi.co.kr/pds/product/2020-03-26_50...   
997  https://ottogi.co.kr/pds/product/2020-03-26_50...   
998  https://ottogi.co.kr/pds/product/2018-06-12_95...   
999  https://ottogi.co.kr/pds/product/2022-07-06_51...   

                                                  link  
0    https://ottogi.co.kr/product/product_view.asp?...  
1    https://ottogi.co.kr/product/product_view.asp?...  
2    https://ottogi.co.kr/product/product_view.asp?...  
3    https://ottogi.co.kr/product/product_view.asp?...  
4    https://ottogi.co.kr/product/product_view.asp?...  
..                                                 ...  
995  https://ottogi.co.kr/product/product_view.asp?...  
996  https://ottogi.co.kr/product/product_view.asp?...  
997  https://ottogi.co.kr/product/product_view.asp?...  
998  https://ottogi.co.kr/product/product_view.asp?...  
999  https://ottogi.co.kr/product/product_view.asp?...  

[1000 rows x 6 columns]

In [ ]:
# 상품명에 포함된 '_'를 ' '으로 바꾸기
df['title'] = df['title'].str.replace('_', ' ')
df

index        title        category  \
0        0          짜슐랭       라면류 > 짜슐랭   
1        1    뿌셔뿌셔 바베큐맛   라면스낵류 > 라면스낵류   
2        2      컵누들 우동맛       라면류 > 컵누들   
3        3      진라면 매운맛   라면류 > 진라면 매운맛   
4        4    쇠고기미역국 라면  라면류 > 쇠고기미역국라면   
..     ...          ...             ...   
995    995     한입쏙쏙 미트볼   레토르트/간편식 > 렌지   
996    996  달콤한 데리야끼 치킨   레토르트/간편식 > 렌지   
997    997      화끈한 닭강정   레토르트/간편식 > 렌지   
998    998  떠먹는 컵피자 불고기  냉장/냉동식품 > 냉동피자   
999    999  헬로베지 채소가득카레   레토르트/간편식 > 3분   

                                                  body  \
0    * 짜슐랭 3스타가이드하나. 고온 로스팅으로 맛을 살린 진한 짜장둘. 파와 양파 기...   
1    1. 바삭하고 고소한 면발에 바베큐맛 스프를 골고루 뿌려서 먹는 부숴 먹는 라면스낵...   
2    1. 120kcal 라는 낮은 칼로리의 제품입니다.2. 담백하고 얼큰한 맛을 즐길 ...   
3              쫄깃하고 부드러운 면발에 진한국물과 얼큰한 양념이 잘 조화된 맛입니다.   
4    쇠고기양지와 사골을 우려낸 진한 육수에 쫄깃하고 부드러운 면발과남해안산 청정미역을 ...   
..                                                 ...   
995  1. 새콤 달콤 맛있는 토마토 소스로 완성한 미트볼, 식이섬유가 풍부하고 두뇌건강에...   
996  1. 달콤하고 짭조름한 데리야끼 소스로 맛있게 조리한 데리야끼 치킨!  2. 달콤짭...   
997  1. 이제는 매운맛이 대세! 매운 고추와 함께 사천요리에 사용되는 두반장과 고추기름...   
998             자연치즈 듬뿍간편하게 떠먹는 컵피자 !누구나 좋아하는 불고기 가득 !   
999  1.비건 브랜드인 HELLO VEGGIE 적용-영국 비건 소사이어티 인증2.8가지 ...   

                                                 image  \
0    https://ottogi.co.kr/pds/product/2022-05-16_10...   
1    https://ottogi.co.kr/pds/product/2023-05-30_64...   
2    https://ottogi.co.kr/pds/product/2023-05-10_22...   
3    https://ottogi.co.kr/pds/product/2022-05-16_10...   
4    https://ottogi.co.kr/pds/product/2020-08-26_55...   
..                                                 ...   
995  https://ottogi.co.kr/pds/product/2022-03-16_15...   
996  https://ottogi.co.kr/pds/product/2020-03-26_50...   
997  https://ottogi.co.kr/pds/product/2020-03-26_50...   
998  https://ottogi.co.kr/pds/product/2018-06-12_95...   
999  https://ottogi.co.kr/pds/product/2022-07-06_51...   

                                                  link  
0    https://ottogi.co.kr/product/product_view.asp?...  
1    https://ottogi.co.kr/product/product_view.asp?...  
2    https://ottogi.co.kr/product/product_view.asp?...  
3    https://ottogi.co.kr/product/product_view.asp?...  
4    https://ottogi.co.kr/product/product_view.asp?...  
..                                                 ...  
995  https://ottogi.co.kr/product/product_view.asp?...  
996  https://ottogi.co.kr/product/product_view.asp?...  
997  https://ottogi.co.kr/product/product_view.asp?...  
998  https://ottogi.co.kr/product/product_view.asp?...  
999  https://ottogi.co.kr/product/product_view.asp?...  

[1000 rows x 6 columns]

In [ ]:
df['title'].value_counts()

옛날 사골곰탕               2
옛날 육개장                2
맛있는 감자떡만두 고기          2
생강차                   2
대추차                   2
                     ..
아임스틱 가볍당 유자차          1
맛있는 참치김치 치즈볶음밥        1
맛있는 참치김치 치즈볶음밥(용기)    1
이천쌀로만든  막걸리식초         1
헬로베지 채소가득카레           1
Name: title, Length: 988, dtype: int64

In [ ]:
# 중복 제거
df = df.drop_duplicates(subset='title', keep='first').reset_index(drop=True)
df.shape

(988, 6)

* 중복을 제거하여 총 988개의 상품 존재

## 필요한 상품만 추출
1. 오뚜기 리뷰 데이터에 존재하는 36개 상품
2. 오뚜기 신제품 44개

### 리뷰 데이터 기반 상품 데이터 추출

In [ ]:
# 리뷰 데이터에 존재하는 36개 상품 추출
product_review = ['맛있는 오뚜기밥',
 '렌지에 돌려먹는 고등어구이 (냉동)',
 '진라면 매운맛',
 '고기리 들기름 막국수',
 '렌지에 돌려먹는 삼치구이',
 '리얼 멕시칸 브리또 페페로니 트리플치즈 (냉동)',
 '옛날 참기름',
 '컵밥 진짬뽕밥',
 '오뚜기밥 큰밥',
 '컵누들 매콤한맛',
 '골드마요네스',
 '토마토케찹',
 '리얼 프렌치 크로크무슈 오리지널 4개입 (냉동)',
 '컵누들 베트남쌀국수',
 '히모카와우동(2인)',
 '열 라면',
 '갈릭고르곤졸라 치즈크러스트 피자',
 '냉동 콤비네이션 피자',
 '마열라면',
 '진라면 순한맛',
 '3분 쇠고기카레',
 '맛있는 큰밥',
 '3분 카레 약간매운맛',
 '맛있는 오뚜기밥 발아현미',
 '맛있는 오뚜기밥 오곡',
 '컵누들 로제맛',
 '진라면컵 매운맛',
 '백세카레 약간매운맛',
 '참깨라면',
 '스낵면',
 '육개장',
 '진비빔면',
 '옛날 구수한 끓여먹는 누룽지',
 '3분 카레 순한맛',
 '옛날 구수한 누룽지',
 '키위 저칼로리 드레싱']

In [ ]:
# 리뷰 데이터에 존재하는 상품만 전체 데이터에서 추출하기
df_review = df[df['title'].isin(product_review)]

In [ ]:
df_review['title'].nunique()

12

In [ ]:
df_review['title'].unique()

array(['진라면 매운맛', '진비빔면', '컵누들 로제맛', '골드마요네스', '마열라면', '옛날 구수한 누룽지',
       '오뚜기밥 큰밥', '3분 쇠고기카레', '렌지에 돌려먹는 삼치구이', '진라면 순한맛', '스낵면', '옛날 참기름'],
      dtype=object)

* 12개만 매핑됨.. 매핑되지 않은 24개 찾아내기

In [ ]:
 [value for value in product_review if value not in df_review['title'].unique()]

['맛있는 오뚜기밥',
 '렌지에 돌려먹는 고등어구이 (냉동)',
 '고기리 들기름 막국수',
 '리얼 멕시칸 브리또 페페로니 트리플치즈 (냉동)',
 '컵밥 진짬뽕밥',
 '컵누들 매콤한맛',
 '토마토케찹',
 '리얼 프렌치 크로크무슈 오리지널 4개입 (냉동)',
 '컵누들 베트남쌀국수',
 '히모카와우동(2인)',
 '열 라면',
 '갈릭고르곤졸라 치즈크러스트 피자',
 '냉동 콤비네이션 피자',
 '맛있는 큰밥',
 '3분 카레 약간매운맛',
 '맛있는 오뚜기밥 발아현미',
 '맛있는 오뚜기밥 오곡',
 '진라면컵 매운맛',
 '백세카레 약간매운맛',
 '참깨라면',
 '육개장',
 '옛날 구수한 끓여먹는 누룽지',
 '3분 카레 순한맛',
 '키위 저칼로리 드레싱']

* 하나하나 매핑하기 -> 리뷰 데이터 상품명을 오뚜기 사이트 상품명 기준으로 변경

In [ ]:
product_review[product_review.index('맛있는 오뚜기밥')] = '오뚜기밥'
product_review[product_review.index('렌지에 돌려먹는 고등어구이 (냉동)')] = '렌지에 돌려먹는 고등어구이 '
product_review[product_review.index('고기리 들기름 막국수')] = '고기리 들기름막국수'
product_review[product_review.index('리얼 멕시칸 브리또 페페로니 트리플치즈 (냉동)')] = 'Real 멕시칸 브리또 페페로니트리플치즈'
product_review[product_review.index('컵밥 진짬뽕밥')] = '맛있는 오뚜기 컵밥 진짬뽕밥'
product_review[product_review.index('컵누들 매콤한맛')] = '컵누들 매콤한 맛'
product_review[product_review.index('토마토케찹')] = '토마토케챂'
product_review[product_review.index('리얼 프렌치 크로크무슈 오리지널 4개입 (냉동)')] = 'REAL 프렌치 크로크무슈 오리지널(4개입)'
product_review[product_review.index('컵누들 베트남쌀국수')] = '컵누들 베트남쌀국수 컵'
product_review[product_review.index('히모카와우동(2인)')] = '히모카와우동(2인) 396G'
product_review[product_review.index('열 라면')] = '열라면'
product_review[product_review.index('갈릭고르곤졸라 치즈크러스트 피자')] = '갈릭고르곤졸라 치즈크러스트 피자 460G'
product_review[product_review.index('냉동 콤비네이션 피자')] = '콤비네이션 피자'
product_review[product_review.index('맛있는 큰밥')] = '오뚜기밥 큰밥'
product_review[product_review.index('3분 카레 약간매운맛')] = '3분카레(약간매운맛)'
product_review[product_review.index('맛있는 오뚜기밥 발아현미')] = '오뚜기밥 발아현미'
product_review[product_review.index('맛있는 오뚜기밥 오곡')] = '오뚜기밥 오곡 '
product_review[product_review.index('진라면컵 매운맛')] = '오뚜기 진라면 매운맛'
product_review[product_review.index('백세카레 약간매운맛')] = '백세카레 (약간매운맛)'
product_review[product_review.index('참깨라면')] = '참깨라면컵'
product_review[product_review.index('육개장')] = '옛날 육개장'
product_review[product_review.index('옛날 구수한 끓여먹는 누룽지')] = '옛날 구수한끓여먹는누룽지(국산)'
product_review[product_review.index('3분 카레 순한맛')] = '3분카레(순한맛)'
product_review[product_review.index('키위 저칼로리 드레싱')] = '키위 저칼로리드레싱(냉장) 310G'

In [ ]:
# 다시 리뷰 데이터에 존재하는 상품만 전체 데이터에서 추출하기
df_review = df[df['title'].isin(product_review)]

In [ ]:
df_review['title'].nunique()

35

In [ ]:
df_review['title'].value_counts()

진라면 매운맛                     1
Real 멕시칸 브리또 페페로니트리플치즈      1
오뚜기밥 큰밥                     1
오뚜기밥 오곡                     1
3분 쇠고기카레                    1
오뚜기밥 발아현미                   1
렌지에 돌려먹는 삼치구이               1
진라면 순한맛                     1
렌지에 돌려먹는 고등어구이              1
3분카레(순한맛)                   1
열라면                         1
스낵면                         1
오뚜기 진라면 매운맛                 1
맛있는 오뚜기 컵밥 진짬뽕밥             1
참깨라면컵                       1
옛날 육개장                      1
오뚜기밥                        1
컵누들 베트남쌀국수 컵                1
진비빔면                        1
토마토케챂                       1
컵누들 로제맛                     1
컵누들 매콤한 맛                   1
콤비네이션 피자                    1
골드마요네스                      1
백세카레 (약간매운맛)                1
3분카레(약간매운맛)                 1
REAL 프렌치 크로크무슈 오리지널(4개입)    1
옛날 구수한 누룽지                  1
히모카와우동(2인) 396G             1
키위 저칼로리드레싱(냉장) 310G         1
옛날 구수한끓여먹는누룽지(국산)           1
갈릭고르곤졸라 치즈크러스트 피자 460G      1
고기리 들기름막국수                  1
마열라면      

### 리뷰데이터 기반 상품 데이터 저장

In [ ]:
df_review

index                     title           category  \
3        3                   진라면 매운맛      라면류 > 진라면 매운맛   
11      11                      진비빔면         라면류 > 진비빔면   
30      30                   컵누들 로제맛          라면류 > 컵누들   
44      44                 컵누들 매콤한 맛          라면류 > 컵누들   
45      45                  콤비네이션 피자     냉장/냉동식품 > 냉동피자   
51      51                    골드마요네스     조미소스/식품 > 마요네스   
52      52              백세카레 (약간매운맛)       분말/건조식품 > 카레   
56      56               3분카레(약간매운맛)      레토르트/간편식 > 3분   
59      59                     토마토케챂       조미소스/식품 > 케챂   
85      85  REAL 프렌치 크로크무슈 오리지널(4개입)    냉장/냉동식품 > 냉동스낵류   
119    119           히모카와우동(2인) 396G     냉장/냉동식품 > 냉장면류   
140    140       키위 저칼로리드레싱(냉장) 310G  냉장/냉동식품 > 냉장/냉동식품   
190    190         옛날 구수한끓여먹는누룽지(국산)       곡류가공식품 > 누룽지   
244    244    갈릭고르곤졸라 치즈크러스트 피자 460G     냉장/냉동식품 > 냉동피자   
298    298                고기리 들기름막국수            면류 > 국수   
308    308                      마열라면         라면류 > 마열라면   
398    399                옛날 구수한 누룽지       곡류가공식품 > 누룽지   
412    413              컵누들 베트남쌀국수 컵          라면류 > 컵누들   
463    464                 3분카레(순한맛)      레토르트/간편식 > 3분   
471    472                      오뚜기밥      곡류가공식품 > 오뚜기밥   
472    473                   오뚜기밥 큰밥      곡류가공식품 > 오뚜기밥   
533    536                  오뚜기밥 오곡       곡류가공식품 > 오뚜기밥   
581    585                  3분 쇠고기카레      레토르트/간편식 > 3분   
589    593                 오뚜기밥 발아현미      곡류가공식품 > 오뚜기밥   
599    603             렌지에 돌려먹는 삼치구이  냉장/냉동식품 > 냉동용기간편식   
624    628                   진라면 순한맛      라면류 > 진라면 순한맛   
788    794    Real 멕시칸 브리또 페페로니트리플치즈    냉장/냉동식품 > 냉동스낵류   
800    806           렌지에 돌려먹는 고등어구이   냉장/냉동식품 > 냉동용기간편식   
804    810                       열라면          라면류 > 열라면   
805    811                       스낵면          라면류 > 스낵면   
810    816               오뚜기 진라면 매운맛      라면류 > 진라면 매운맛   
871    877           맛있는 오뚜기 컵밥 진짬뽕밥      레토르트/간편식 > 컵밥   
882    888                     참깨라면컵         라면류 > 참깨라면   
920    929                    옛날 육개장   레토르트/간편식 > 탕국찌개류   
949    961                    옛날 참기름         식용유지 > 참기름   

                                                  body  \
3              쫄깃하고 부드러운 면발에 진한국물과 얼큰한 양념이 잘 조화된 맛입니다.   
11   - 배/사과/매실/무의 새콤달콤한 맛, 태양초의 매운맛, 타마린드양념소스의 시원 새...   
30   꾸덕한 로제떡볶이 소스와쫄깃탱글한 당면의 만남! 매일같이 배달시켜 먹던 K-로제 완...   
44   1. 120kcal라는 낮은 칼로리의 제품입니다.2. 맵고 칼칼한 맛을 즐길 수 있...   
45   고소한 모짜렐라 치즈와 고기, 야채가 잘 조화된 먹음직스러운 콤비네이션 피자입니다....   
51        고소하고 부드러운 대한민국 1등 마요네스!진하고 고소한 맛을 즐기실수 있습니다.   
52   1. 강황함량이 57.4% 더 증가(오뚜기카레 약간매운맛 기준)한 고급카레입니다.2...   
56   3분 카레는? 오뚜기 3분 카레는 향긋한 열대지방의 천연향신료로 만든 과일소스와 국...   
59   1. 오뚜기 케챂은 진한 케챂으로 우리 입맛에 맞는 우리의 자랑 대한민국 대표 케챂...   
85                            -오리지널 프렌치 스타일의 햄&치즈 샌드위치   
119  - 트렌디한 메뉴를 가정에서 즐겨요!- 독특한 식감으로 먹는 재미 UP!- 장국에 ...   
140  '키위퓨레의 새콤달콤한 맛과 어우러져 산뜻하게 맛있는 키위 저칼로리 드레싱과일샐러드...   
190  - 불린 쌀을 굽지않고, 갓 지은 밥을 무쇠판에 직접 구워내어더욱 구수합니다.- 구...   
244    '블루치즈' 고르곤졸라치즈 쏙쏙, 쭈욱 늘어나는 체다치즈가 가득한 맛있는 크러스트피자   
298  ·담백하고 고소하면서도 평양냉면처럼 자꾸자꾸 생각나는 슴슴하면서 중독적인 맛·오뚜기...   
308  마늘, 후추, 고추 3가지 매운맛의 조화로 깊고 맛있게 매운라면*특징: 제주 대정마...   
398  1. 옛날 맛 그대로 무쇠판에 직접 구워내어 구수한 누룽지의 깊은 맛을 즐기실 수 ...   
412  1. 튀기지 않은 쌀면을 사용하여 부담없이 즐기실 수 있습니다.2. 쇠고기 육수와 ...   
463  3분 카레는? 오뚜기 3분 카레는 향긋한 열대지방의 천연 향신료로 만든 카레분과 국...   
471  1. 오뚜기밥은 품질 좋은 벼를 저온창고에 보관했다가 필요한 만큼만 도정하여 정성껏...   
472  1. 오뚜기밥은 품질 좋은 벼를 저온창고에 보관했다가 필요한 만큼만 도정하여 정성껏...   
533  1. 평소 집에서 즐겨먹는 쌀, 찹쌀, 찰현미, 찰흑미, 차수수 등 5가지의 곡물을...   
581  3분 쇠고기카레는? 오뚜기 3분 쇠고기카레는 열대지방의 천연향신료로 만든 카레분과 ...   
589  1. 발아현미는 쌀의 눈에 있는 영양소를 그대로 함유하고 있는 현미를 발아시켜 효소...   
599  1. 해동없이 전자레인지로 2분(700W기준)만 돌려 간편하게 먹는 렌지용 생선구이...   
624  1. 오뚜기 진라면 순한맛은 쫄깃하고 찰기있는 면에 사골국물과 감칠맛이 조화된 진하...   
788                         페페로니 햄과 모짜렐라, 체다, 고다치즈가 듬뿍   
800  1. 해동없이 전자레인지로 2분(700W기준)만 돌려 간편하게 먹는 렌지용 생선구이...   
804            쫄깃하고 찰기있는 면에 하늘초의 매콤함이 어우러져 가장 화끈하게 매운맛   
805  2분만에 조리 되는 얇고 꼬들꼬들한 면과 진한 쇠고기 국물의 담백한 맛이 특징인 라...   
810           쫄깃하고 부드러운 면발에 진한 국물과 얼큰한 양념이 잘 조화된 맛입니다.   
871  - 집밥처럼 맛있게!- 맛도 편리함

In [ ]:
df_review.to_csv("df_review.csv", encoding = "utf-8-sig")

* 리뷰기반 상품 데이터 총 35개

### 신제품 상품만 추출

In [ ]:
# 신제품 44개 상품 추출
product_new = ['이천쌀로만든 막걸리식초', '청송사과로만든 순사과식초', '가뿐한끼 새우계란곤약볶음밥', '가뿐한끼 쇠고기곤약볶음밥',
               '가뿐한끼 매콤닭가슴살곤약볶음밥', '가뿐한끼 닭가슴살곤약볶음밥', '마열라면 용기', '부산식 얼큰돼지국밥',
               '병천식 얼큰순대국밥', '꼬리까지 가득 찬 슈크림붕어빵', '꼬리까지 가득 찬 팥붕어빵', '맛있는 베이컨김치볶음밥',
               '맛있는 계란볶음밥','마열라면', 'X.O. 감자떡만두 김치', 'X.O. 감자떡만두 고기',
               '오즈키친 레드칠리 칰 윙봉', '오즈키친 스윗허니 치킨 윙봉', '직접볶은 트러플간짜장 고형', '콘치즈피자',
               '히모카와우동', '참깨라면볶음면', '[위글위글콜라보]사과쨈', '맛있는 빠삭 튀김만두 땡초',
               '맛있는 빠삭 튀김만두 고기', '초계 물막국수', '튀만전찍먹소스', 'BIG컵누들 우동맛 용기',
               'BIG컵누들 매콤한맛 용기', '오뚜기밥 수향미밥', '베이컨크림 뇨끼', '트러플크림 뇨끼',
               '키위 저칼로리 드레싱', '삽겹살 와사비 고추장소스', '삽겹살 제주식멜젓소스', '삼겹살 양파절임소스',
               '순후추 떡볶이', '컵밥 오뚜기 카레', '컵밥 참기름김치볶음밥', '직접 갈아먹는 스모크페퍼솔트',
               '제주담은 제주 감귤도우 피자', '황도(반절)', '제주똣똣라면', '양송이컵스프(컵)']

In [ ]:
# 신제품 데이터에 존재하는 상품만 전체 데이터에서 추출하기
df_new = df[df['title'].isin(product_new)]

In [ ]:
df_new['title'].nunique()

3

* 3개만 매핑됨.. 매핑되지 않은 41개 찾아내기

In [ ]:
 [value for value in product_new if value not in df_new['title'].unique()]

['이천쌀로만든 막걸리식초',
 '청송사과로만든 순사과식초',
 '가뿐한끼 새우계란곤약볶음밥',
 '가뿐한끼 쇠고기곤약볶음밥',
 '가뿐한끼 매콤닭가슴살곤약볶음밥',
 '가뿐한끼 닭가슴살곤약볶음밥',
 '마열라면 용기',
 '부산식 얼큰돼지국밥',
 '병천식 얼큰순대국밥',
 '꼬리까지 가득 찬 슈크림붕어빵',
 '꼬리까지 가득 찬 팥붕어빵',
 '맛있는 베이컨김치볶음밥',
 '맛있는 계란볶음밥',
 'X.O. 감자떡만두 김치',
 'X.O. 감자떡만두 고기',
 '오즈키친 레드칠리 칰 윙봉',
 '오즈키친 스윗허니 치킨 윙봉',
 '직접볶은 트러플간짜장 고형',
 '콘치즈피자',
 '히모카와우동',
 '참깨라면볶음면',
 '[위글위글콜라보]사과쨈',
 '맛있는 빠삭 튀김만두 땡초',
 '맛있는 빠삭 튀김만두 고기',
 '초계 물막국수',
 '튀만전찍먹소스',
 'BIG컵누들 우동맛 용기',
 'BIG컵누들 매콤한맛 용기',
 '오뚜기밥 수향미밥',
 '베이컨크림 뇨끼',
 '트러플크림 뇨끼',
 '키위 저칼로리 드레싱',
 '삽겹살 와사비 고추장소스',
 '삽겹살 제주식멜젓소스',
 '순후추 떡볶이',
 '컵밥 오뚜기 카레',
 '컵밥 참기름김치볶음밥',
 '직접 갈아먹는 스모크페퍼솔트',
 '제주담은 제주 감귤도우 피자',
 '황도(반절)',
 '양송이컵스프(컵)']

* 하나하나 매핑하기 -> 신제품 데이터 상품명을 오뚜기 사이트 상품명 기준으로 변경

In [ ]:
product_new[product_new.index('이천쌀로만든 막걸리식초')] = '이천쌀로만든  막걸리식초'
product_new[product_new.index('청송사과로만든 순사과식초')] = '청송사과로만든  순사과식초'
# 가뿐한끼 4개 제품은 오뚜기 사이트에 존재 X
product_new[product_new.index('마열라면 용기')] = '마열라면'
#  '부산식 얼큰돼지국밥', '병천식 얼큰순대국밥' 2개 존재 X
product_new[product_new.index('꼬리까지 가득 찬 슈크림붕어빵')] = '꼬리까지 가득 찬 슈크림붕어빵 480G'
product_new[product_new.index('꼬리까지 가득 찬 팥붕어빵')] = '꼬리까지 가득 찬 팥붕어빵 480G'
product_new[product_new.index('맛있는 베이컨김치볶음밥')] = '맛있는 베이컨김치볶음밥(원팩) 450G'
product_new[product_new.index('맛있는 계란볶음밥')] = '맛있는 계란볶음밥(원팩) 450G'
product_new[product_new.index('X.O. 감자떡만두 김치')] = 'X.O. 감자떡만두 김치 (320gX2)'
product_new[product_new.index('X.O. 감자떡만두 고기')] = 'X.O. 감자떡만두 고기 (320gX2)'
product_new[product_new.index('오즈키친 레드칠리 칰 윙봉')] = '오즈키친 레드칠리 칰 윙&봉'
product_new[product_new.index('오즈키친 스윗허니 치킨 윙봉')] = '오즈키친 스윗허니 칰 윙&봉'
product_new[product_new.index('직접볶은 트러플간짜장 고형')] = '직접 볶은 트러플 간짜장 고형'
product_new[product_new.index('콘치즈피자')] = '콘치즈 피자'
product_new[product_new.index('히모카와우동')] = '히모카와우동(2인) 396G'
product_new[product_new.index('참깨라면볶음면')] = '참깨라면 볶음면'
# '[위글위글콜라보]사과쨈' 1개 존재 X
product_new[product_new.index('맛있는 빠삭 튀김만두 땡초')] = '빠삭 튀김만두 땡초 (334G*2)'
product_new[product_new.index('맛있는 빠삭 튀김만두 고기')] = '빠삭 튀김만두 고기 (334G*2)'
product_new[product_new.index('초계 물막국수')] = '초계 물막국수(2인) 1,034G'
product_new[product_new.index('튀만전찍먹소스')] = '튀만전찍먹소스 320G'
product_new[product_new.index('BIG컵누들 우동맛 용기')] = '빅컵누들 우동맛'
product_new[product_new.index('BIG컵누들 매콤한맛 용기')] = '빅컵누들 매콤한맛'
product_new[product_new.index('오뚜기밥 수향미밥')] = '오뚜기밥 수향미'
product_new[product_new.index('베이컨크림 뇨끼')] = '베이컨크림뇨끼'
product_new[product_new.index('트러플크림 뇨끼')] = '트러플크림뇨끼'
product_new[product_new.index('키위 저칼로리 드레싱')] = '키위 저칼로리드레싱(냉장) 310G'
product_new[product_new.index('삽겹살 와사비 고추장소스')] = '삼겹살 와사비 고추장소스 290G' # 용량 여러개 존재
product_new[product_new.index('삽겹살 제주식멜젓소스')] = '삼겹살 제주식 멜젓소스 270g' # 용량 여러개 존재
product_new[product_new.index('순후추 떡볶이')] = '순후추떡볶이'
# 컵밥 오뚜기 카레, 컵밥 참기름김치볶음밥 2개 존재 X
product_new[product_new.index('직접 갈아먹는 스모크페퍼솔트')] = '직접 갈아먹는 스모크페퍼솔트 70g'
product_new[product_new.index('제주담은 제주 감귤도우 피자')] = '제주담음 제주감귤도우피자'
product_new[product_new.index('황도(반절)')] = '황도 통조림'
# 양송이 컵스프 1개 존재 X

In [ ]:
# 다시 신제품 데이터에 존재하는 상품만 전체 데이터에서 추출하기
df_new = df[df['title'].isin(product_new)]

In [ ]:
df_new['title'].nunique()

33

In [ ]:
 [value for value in product_new if value not in df_new['title'].unique()]

['가뿐한끼 새우계란곤약볶음밥',
 '가뿐한끼 쇠고기곤약볶음밥',
 '가뿐한끼 매콤닭가슴살곤약볶음밥',
 '가뿐한끼 닭가슴살곤약볶음밥',
 '부산식 얼큰돼지국밥',
 '병천식 얼큰순대국밥',
 '[위글위글콜라보]사과쨈',
 '컵밥 오뚜기 카레',
 '컵밥 참기름김치볶음밥',
 '양송이컵스프(컵)']

### 신제품 데이터 기반 상품 데이터 저장

In [ ]:
df_new

index                   title           category  \
46      46       삼겹살 제주식 멜젓소스 270g     조미소스/식품 > 육류소스   
69      69                  제주똣똣라면       라면류 > 제주똣똣라면   
116    116                빅컵누들 우동맛          라면류 > 컵누들   
119    119         히모카와우동(2인) 396G     냉장/냉동식품 > 냉장면류   
120    120        직접 볶은 트러플 간짜장 고형     분말/건조식품 > 분말짜장   
124    124           제주담음 제주감귤도우피자     냉장/냉동식품 > 냉동피자   
140    140     키위 저칼로리드레싱(냉장) 310G  냉장/냉동식품 > 냉장/냉동식품   
166    166      초계 물막국수(2인) 1,034G     냉장/냉동식품 > 냉장면류   
170    170         오즈키친 스윗허니 칰 윙&봉    냉장/냉동식품 > 조리냉동류   
171    171   꼬리까지 가득 찬 슈크림붕어빵 480G     냉장/냉동식품 > 냉동빵류   
177    177                  순후추떡볶이       곡류가공식품 > 쌀떡류   
207    207     직접 갈아먹는 스모크페퍼솔트 70g       향신료류 > 고급향신료   
211    211               빅컵누들 매콤한맛          라면류 > 컵누들   
224    224            튀만전찍먹소스 320G     조미소스/식품 > 육류소스   
270    270         오즈키친 레드칠리 칰 윙&봉    냉장/냉동식품 > 조리냉동류   
308    308                    마열라면         라면류 > 마열라면   
331    331                오뚜기밥 수향미    카테고리없음 > 카테고리없음   
332    332      맛있는 계란볶음밥(원팩) 450G      냉장/냉동식품 > 냉동밥   
339    339           이천쌀로만든  막걸리식초       조미소스/식품 > 식초   
340    340   맛있는 베이컨김치볶음밥(원팩) 450G      냉장/냉동식품 > 냉동밥   
361    361              삼겹살 양파절임소스     조미소스/식품 > 육류소스   
470    471                  황도 통조림  농/축/수산식품 > 과일통조림류   
704    710      삼겹살 와사비 고추장소스 290G     조미소스/식품 > 육류소스   
706    712  X.O. 감자떡만두 고기 (320gX2)       냉장/냉동식품 > 만두   
712    718  X.O. 감자떡만두 김치 (320gX2)       냉장/냉동식품 > 만두   
742    748          청송사과로만든  순사과식초       조미소스/식품 > 식초   
777    783                 트러플크림뇨끼    냉장/냉동식품 > 조리냉동류   
778    784                참깨라면 볶음면         라면류 > 참깨라면   
845    851                 베이컨크림뇨끼  냉장/냉동식품 > 냉장/냉동식품   
904    910     빠삭 튀김만두 고기 (334G*2)       냉장/냉동식품 > 만두   
907    913     빠삭 튀김만두 땡초 (334G*2)       냉장/냉동식품 > 만두   
966    978                  콘치즈 피자     냉장/냉동식품 > 냉동피자   
969    981     꼬리까지 가득 찬 팥붕어빵 480G     냉장/냉동식품 > 냉동빵류   

                                                  body  \
46   유명한 제주 고깃집에서 먹던 바로 그 멜젓소스를 가정에서나 캠핑장에서 즐길 수 있다...   
69   1. 제주 라면 맛집 ‘금악똣똣라면’과 진라면의 특급 콜라보  - 진라면과 금악똣똣...   
116  이제 컵누들도 BIG으로 가볍고 든든하게 즐기자!스테디셀러 컵누들 우동맛 소컵을 그...   
119  - 트렌디한 메뉴를 가정에서 즐겨요!- 독특한 식감으로 먹는 재미 UP!- 장국에 ...   
120  고급스러운 트러플과 직접 볶은 춘장의 풍미가 조화로운 구수한 짜장!1인분씩 소포장되...   
124  ·제주 감귤절임을 넣어 향긋한 감귤도우·제주산 당근, 적양배추, 양배추로 만든 상큼...   
140  '키위퓨레의 새콤달콤한 맛과 어우러져 산뜻하게 맛있는 키위 저칼로리 드레싱과일샐러드...   
166  - 추가 고명 없이도 완전한 '한' 그릇이 완성되는 프리미엄 초계 물막국수!- 초계...   
170  에어프라이어 간단 조리로 갓 튀긴 듯 바삭한 치킨을 특제 스윗허니소스와 함께 집에서...   
171  - 마지막 한 입까지 맛있게!머리부터 꼬리까지 버리는 곳 없이 속을 꽉~ 채운 - ...   
177  얼얼하고 강렬한 후추맛이 특징인1등 순후추로 만든 순후추떡볶이순후추와 가장 잘 어울...   
207  간편하게 뿌려서 더하는 근사한 훈제풍미!통후추와 암염을 '천연 너도밤나무'로 직접 ...   
211  이제 컵누들도 BIG으로 가볍고 든든하게 즐기자!스테디셀러 컵누들 매콤한맛 소컵을 ...   
224  국산 현미로 만든 흑초로, 튀김, 만두, 전 등 기름진 요리의 느끼함을 잡아주는 새...   
270  에어프라이어 간단 조리로 갓 튀긴 듯 바삭한 치킨을 특제 레드칠리소스와 함께 집에서...   
308  마늘, 후추, 고추 3가지 매운맛의 조화로 깊고 맛있게 매운라면*특징: 제주 대정마...   
331  은은한 누룽지향과 찰진 식감국내에서 개발되었고 화성시 대표쌀 품종인 '골든퀸3호'로...   
332  한국인의 소울푸드를 어디서나 쉽고 간편하게!   - 고슬고슬하고 담백한 볶음밥을 어...   
339  이천을 대표하는 특산물 쌀과 오뚜기의 46년 식초 제조 노하우의 만남- 두번의 발효...   
340  한국인의 소울푸드를 어디서나 쉽고 간편하게!   - 고슬고슬하고 담백한 볶음밥을 어...   
361  - 채썬 양파에 부어 구운 고기를 찍어먹는 액상소스- 사과와 배를 이용해 달콤하고 ...   
470  1. 엄선된 좋은 복숭아만을 들여와 국내에서 가공한 제품으로 신선하며,?맛과 향이 ...   
704  “고추장과 와사비의 예상치 못한 조합!”새롭게 떠오르는 고기 소스계의 신예, 와사비...   
706  얇고 투명한 감자떡같은 쫄깃한 만두피에 100% 국산 돼지고기와 야채를 듬뿍 넣어 ...   
712  얇고 투명한 감자떡같은 쫄깃한 만두피에 100% 국산 돼지고기와 칼칼한 김치를 넣은...   
742  청송을 대표하는 특산물 사과와 오뚜기의 46년 식초 제조 노하우의 만남- 두번의 발...   
777  이탈리아산 트러플 엑스트라 버진 오일을 사용해 진짜 트러플향을 느낄 수 있는 트러플...   
778  1. 참기름 향이 솔솔 나는 매콤 고소한 ‘참깨라면’의 볶음면 버전2. '참깨라면'...   
845           바삭바삭 리얼하게 구운 베이컨칩이 들어있어 식감마저 풍부한 베이컨크림뇨끼   
904  기존 군만두보다 한번 더 튀겨서 더욱 빠삭한 식감으로, 에어프라이어로 단 6분만에 ...   
907  기존 군만두보다 한번 더 튀겨서 더욱 빠삭한 식감으로, 에어프라이어로 단 6분만에 ...   
966  - 쫄깃한 도우의 스톤베이크드 피자! 생 이스트로 반죽한 숙성 도우를 돌판 오븐에 ...   
969  - 마지막 한 입까지 맛있게!머리부터 꼬리까지 버리는 곳 없이 속을 꽉~ 채운 - ...   

                                                 image  \
46   https://ottogi.co.kr/pds/produ

In [ ]:
df_new.to_csv("df_new.csv", encoding = "utf-8-sig")

* 신제품 기반 상품 데이터 총 33개